src/main/java/racing/RacingGame.java

```diff
@@ -12,30 +13,51 @@
 public class RacingGame {
 
     private MoveStrategy moveStrategy;
+    private RacingCarList racingCarList;
+    private String[] nameOfRacingCarList;
+    private int countOfRacing;
+
+    public RacingGame() {}
+    public RacingGame(MoveStrategy moveStrategy, String[] nameOfRacingCarList, int countOfRacing) {
+        this.moveStrategy = moveStrategy;
+        this.nameOfRacingCarList = nameOfRacingCarList;
+        this.countOfRacing = countOfRacing;
+    }
+
+    public RacingGame(MoveStrategy moveStrategy) {
+        this.moveStrategy = moveStrategy;
+    }
 
     public static void main(String[] args) {
-        int countOfRacingCar = InputView.getRacingCarInput();
+        String[] nameOfRacingCarList = InputView.getRacingCarInputWithName();
         int countOfRacing = InputView.getRacingInput();
+        MoveStrategy moveStrategy = new RacingMoveStategy();
+        RacingGame racingGame = new RacingGame(moveStrategy, nameOfRacingCarList, countOfRacing);
+
+        racingGame.start();
 
-        MoveStrategy moveStategy = new RacingMoveStategy();
-        RacingGame racingGame = new RacingGame(moveStategy);
-        List<RacingCar> racingCarList = racingGame.setRacing(countOfRacing);
-        List<RacingCar> racingResult = racingGame.startRacing(countOfRacing, racingCarList);
-        System.out.println(racingResult);
-        ResultView.viewRacingResult(racingResult);
+        ResultView.viewRacingResult(racingGame.getRacingCarList());
     }
-    public RacingGame() {}
 
-    public RacingGame(MoveStrategy moveStrategy) {
-        this.moveStrategy = moveStrategy;
+    private void start() {
+        this.racingCarList = new RacingCarList(nameOfRacingCarList, moveStrategy);
+        racingCarList.startRacing(countOfRacing);
     }
 
+    private RacingCarList getRacingCarList() {
+        return racingCarList;
+    }
+
+
     public List<RacingCar> setRacing(int countOfRacingCar) {
-       return Stream.generate(() -> new RacingCar(moveStrategy)).limit(countOfRacingCar).collect(Collectors.toList());
+       return Stream.generate(() -> new RacingCar(moveStrategy)).limit(countOfRacingCar)
+                                                                .collect(Collectors.toList());
```

Comment:
```suggestion
       return Stream.generate(() -> new RacingCar(moveStrategy))
                    .limit(countOfRacingCar)
                    .collect(Collectors.toList());

```
stream method는 한 줄에 하나만 사용해주세요!